src/flows/context/flow.current.tsx

```diff
@@ -69,37 +69,45 @@ export const FlowProvider: FC = ({children}) => {
   }, [flows, currentID])
 
   useEffect(() => {
-    if (isFlagEnabled('sharedFlowEditing')) {
+    const doc = yDoc.current
+    if (isFlagEnabled('sharedFlowEditing') && currentID) {
       provider.current = new WebsocketProvider(
         'wss://demos.yjs.dev', // todo(ariel): replace this with an actual API that we setup
-        flowId, // todo(ariel): we might need to confine this to an org, depends on how multi-org plays out
-        yDoc.current
+        currentID, // todo(ariel): we might need to confine this to an org, depends on how multi-org plays out
+        doc
       )
       provider.current.on('sync', isSynced => {
         if (isSynced) {
-          const localState = yDoc.current.getMap('localState')?.toJSON()
-          if (!localState) {
-            yDoc.current.getMap('localState').set('localState', currentFlow)
+          const {flowUpdateData} = doc.getMap('flowUpdateData')?.toJSON()
+          if (!flowUpdateData && currentFlow) {
+            doc
+              .getMap('flowUpdateData')
+              .set('flowUpdateData', serialize(currentFlow, orgID))
           }
         }
       })
-      yDoc.current.on('update', () => {
-        const {localState} = yDoc.current.getMap('localState').toJSON()
-        setCurrentFlow(prev => {
-          if (btoa(JSON.stringify(prev)) === btoa(JSON.stringify(localState))) {
-            return prev
-          }
-          return localState
-        })
+    }
+
+    const onUpdate = () => {
+      const {flowUpdateData} = doc.getMap('flowUpdateData').toJSON()
+      const hydrated = hydrate(flowUpdateData?.data)
+
+      setCurrentFlow(prev => {
+        if (btoa(JSON.stringify(prev)) === btoa(JSON.stringify(hydrated))) {
+          return prev
+        }
+        return hydrated
       })
     }
 
+    doc.on('update', onUpdate)
     return () => {
       if (isFlagEnabled('sharedFlowEditing')) {
         disconnectProvider()
       }
+      doc.off('update', onUpdate)
     }
-  }, [flowId])
+  }, [currentID, currentFlow, orgID])
```

Comment:
remove the orgid, and how often does the disconnect / reconnect logic fire when you're editing a flux script?

Reply:
It's pretty slow with the `orgID` included, but in both cases there's a bunch of syncing and disconnecting